rust/Backtracking/combinations_of_sum_k.rs

```diff
@@ -0,0 +1,41 @@
+pub fn combinations_of_sum_k(nums: Vec<i32>, target: i32) -> Vec<Vec<i32>> {
+    let mut result: Vec<Vec<i32>> = Vec::new();
+    let mut combination: Vec<i32> = Vec::new();
+    
+    combinations_of_sum_k_impl(&mut combination, 0, &nums, target, &mut result);
+    
+    result
```

Comment:
Please use the same variable/function naming as the python solutions

Reply:
Same for the `xxx_impl` functions

Reply:
Done

Reply:
also `result` -> `res`

Reply:
Done, and updated a few other variables to match python as well.